component/src/main/java/io/siddhi/extension/io/cdc/source/listening/MongoChangeDataCapture.java

```diff
@@ -0,0 +1,137 @@
+
+        //Map to return
+        Map<String, Object> detailsMap = new HashMap<>();
+
+        Struct record = (Struct) connectRecord.value();
+
+        //get the change data object's operation.
+        String op;
+
+        try {
+            op = (String) record.get(CDCSourceConstants.CONNECT_RECORD_OPERATION);
+        } catch (NullPointerException | DataException ex) {
+            return detailsMap;
+        }
+
+        //match the change data's operation with user specifying operation and proceed.
+        if (operation.equalsIgnoreCase(CDCSourceConstants.INSERT) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_INSERT_OPERATION)
+                || operation.equalsIgnoreCase(CDCSourceConstants.DELETE) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_DELETE_OPERATION)
+                || operation.equalsIgnoreCase(CDCSourceConstants.UPDATE) &&
+                op.equals(CDCSourceConstants.CONNECT_RECORD_UPDATE_OPERATION)) {
+
+            switch (op) {
+                case CDCSourceConstants.CONNECT_RECORD_INSERT_OPERATION:
+                    //append row details after insert.
+                    String insertString = (String) record.get(CDCSourceConstants.AFTER);
+                    JSONObject jsonObj = new JSONObject(insertString);
+                    detailsMap = getMongoDetailMap(jsonObj);
+                    break;
+                case CDCSourceConstants.CONNECT_RECORD_DELETE_OPERATION:
+                    //append row details before delete.
+                    String deleteDocumentId = (String) ((Struct) connectRecord.key())
+                            .get(CDCSourceConstants.MONGO_COLLECTION_ID);
+                    JSONObject jsonObjId = new JSONObject(deleteDocumentId);
+                    detailsMap.put(CDCSourceConstants.MONGO_COLLECTION_ID,
+                            jsonObjId.get(CDCSourceConstants.MONGO_COLLECTION_OBJECT_ID));
+
+                    break;
+                case CDCSourceConstants.CONNECT_RECORD_UPDATE_OPERATION:
+                    //append row details before update.
+                    String updateDocument = (String) record.get(CDCSourceConstants.MONGO_PATCH);
+                    JSONObject jsonObj1 = new JSONObject(updateDocument);
+                    JSONObject setJsonObj = (JSONObject) jsonObj1.get(CDCSourceConstants.MONGO_SET);
+                    detailsMap = getMongoDetailMap(setJsonObj);
+                    String updateDocumentId = (String) ((Struct) connectRecord.key())
+                            .get(CDCSourceConstants.MONGO_COLLECTION_ID);
+                    JSONObject jsonObjId1 = new JSONObject(updateDocumentId);
+                    detailsMap.put(CDCSourceConstants.MONGO_COLLECTION_ID,
+                            jsonObjId1.get(CDCSourceConstants.MONGO_COLLECTION_OBJECT_ID));
+                    break;
+                default:
+                    break;
```

Comment:
in the default case shall we add a log saying the op value is not supported ?

Reply:
Added a log in default case.